clang/include/clang/Basic/DiagnosticParseKinds.td

```diff
@@ -299,6 +299,12 @@ def note_missing_selector_name : Note<
 def note_force_empty_selector_name : Note<
   "or insert whitespace before ':' to use %0 as parameter name "
   "and have an empty entry in the selector">;
+def ext_c_label_followed_by_declaration : ExtWarn<
+  "label followed by a declaration is a C23 extension">,
+  InGroup<C23>;
+def warn_c23_compat_label_followed_by_declaration : Warning<
+  "label followed by a declaration is incompatible with C standards before "
+  "C23">, InGroup<CPre23Compat>, DefaultIgnore;
```

Comment:
I can't find tests that would check that these messages are emitted. Are they coming?

Reply:
Good catch, those slipped through the cracks. I'll add the tests shortly.

Reply:
This should be handled now.